scripts/o3de/tests/ExportScripts/test_export_source_android.py

```diff
@@ -0,0 +1,880 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import logging
+import pytest
+import pathlib
+from unittest.mock import patch, create_autospec, MagicMock, Mock, PropertyMock
+from o3de.export_project import O3DEScriptExportContext
+from o3de.command_utils import O3DEConfig
+
+# Because the export scripts are standalone, we have to manually import them to test
+import o3de.utils as utils
+from o3de import android_support
+import platform
+utils.prepend_to_system_path(pathlib.Path(__file__).parent.parent.parent / 'ExportScripts')
+import export_source_android as expa
+
+
```

Comment:
How similar are the test_export_xyz.py scripts?  some of this code looks really similar to pc tests and it's a lot of code, so if there's some common tests we should run on all the scripts lets move that to a `test_export_common.py` and avoid the duplication and runtime cost.

Reply:
They are similar in pattern, but due to argument differences, the exact code is not the same between them for the larger tests, making it difficult to extract into a common script. It's possible the asset bundler/processor related tests are a good starting candidate.

Reply:
the main reason that most of the code is repeated and yet difficult to extract is b/c alot of the time, we are doing either of the following:
1. Calling the export function
2. Mocking the export function

The export function between PC and android is not the same (different argument patterns altogether), and thus  the surrounding logic while similar in structure, still has to slot in to each export func uniquely.